src/sentry/static/sentry/app/components/forms/multipleCheckboxField.tsx

```diff
@@ -7,27 +7,54 @@ import Tooltip from 'app/components/tooltip';
 import {IconQuestion} from 'app/icons';
 import {defined} from 'app/utils';
 
-export default class MultipleCheckboxField extends FormField {
+type Value = string | number | boolean;
+
+type Props = {
+  hideLabelDivider?: boolean;
+  choices: Array<[number | string, number | string]>;
+} & FormField['props'];
+
+type State = FormField['state'] & {
+  values: Value[];
+};
+
+export default class MultipleCheckboxField extends FormField<Props, State> {
   static propTypes = {
     ...FormField.propTypes,
     hideLabelDivider: PropTypes.bool,
     choices: PropTypes.array.isRequired,
   };
 
-  onChange = (value, e) => {
-    let allValues = this.state.value;
-    if (e.target.checked) {
-      if (allValues) {
-        allValues = [...allValues, value];
+  onChange = (e: React.ChangeEvent<HTMLInputElement>, value?: Value) => {
+    let allValues = this.state.values;
+    if (value) {
```

Comment:
If there is no value the state won't be updated. That seems like a behavior change that could be relevant.

Reply:
👍

Reply:
This was from complications with overloading `onChange`, since the overloaded signature has to have optional extended params iirc.